### PR TITLE
Let matrix sdk be used through Maven

### PR DESCRIPTION
**Pull request by vincentbrison**
This PR fix the generated pom.xml when releasing the library using maven, enable the use of Jitpack or any Maven repository.

### CODE REVIEW

**Comment by Matrixcoffee** [comment]
This is related to vector-im/riot-android#1284, correct?

**Comment by vincentbrison** [comment]
It suits the same goal. The purpose of this PR is to let the projet be Maven compatible, thus, when using `maven` plugin for gradle, it will generate valid `pom.xml` with is not the case with the current declaration of `olm-sdk` https://github.com/matrix-org/matrix-android-sdk/pull/175/commits/ab37caa5ce40f507f4674a803d2544504e8715e8.

Then, if you want to release `matrix-android-sdk` on a Maven repository, it will be easy, and even more, we can directly as a developer start using the sdk using maven from Jitpack. Jitpack let you reference commit, github release, branches.

This feature bring a huge enhancement in dependency management, since projets using the SDK will no more include the binary and declare the dependencies of this binary, needing to know from which commit came the binary. Only add the Maven dependency.

**Comment by vincentbrison** [comment]
Signed-off-by: Vincent Brison vincent.brison@fabernovel.com

File: matrix-sdk/build.gradle [hunk 1]
```diff
@@ -100,7 +100,7 @@ dependencies {
     compile 'com.squareup.okhttp:okhttp:2.2.0'
     compile 'io.pristine:libjingle:9690@aar'
 
-    compile(name: 'olm-sdk', ext: 'aar')
+    compile 'org.matrix:olm-sdk:olm-sdk@aar'
```
Comment (bmarty) [inline matrix-sdk/build.gradle]: Unfortunately, this still uses the embedded binary file libs/olm-sdk.aar (if you remove this file it does not work anymore)
Is it the expected behavior?

Comment (bmarty) [reply matrix-sdk/build.gradle]: Note: this is due to the lines in build.gradle
>        flatDir {
>            dir 'libs'
>        }

**Pull request closed without merging**